engine/Pointer.js

```diff
@@ -0,0 +1,43 @@
+import {Path} from 'path.js';
+
+export class Pointer{
+	constructor(container, index){
+		this.container = container;
+		this.index = index;
+	}
+
+	Resolve(){
+		if (index < 0) return this.container;
+		if (this.container == null) return null;
+		if (this.container.content.length == 0) return this.container;
+		if (index >= this.container.content.length) return null;
+
+		return this.container.content[index];
+	}
+
+	get isNull(){
+		return this.container == null;
+	}
+
+	get path(){
+		if (this.isNull()) return null;
```

Comment:
Since is null is a `get`, I believe we want `if (this.isNull)`, ie. not calling the function.

Reply:
Typo again! I wish VS Code would warn me about that stuff, I guess I didn't configure it properly.